CONTRIBUTING.md

```diff
@@ -14,23 +14,31 @@ If you are planning to start some significant coding, you would benefit from ask
 
 1. [Fork](https://help.github.com/articles/fork-a-repo/) the repository and [clone](https://help.github.com/articles/cloning-a-repository/) your fork.
 
-2. Start coding!
+2. Before you start coding, open an [issue at Github](https://github.com/minetest/minetest/issues) to discuss your intended contribution with the core developers, this may save time for you and others, and may avoid wasted work.
+
+3. Start coding!
     - Refer to the [Lua API](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt), [Developer Wiki](http://dev.minetest.net/Main_Page) and other [documentation](https://github.com/minetest/minetest/tree/master/doc).
     - Follow the [C/C++](http://dev.minetest.net/Code_style_guidelines) and [Lua](http://dev.minetest.net/Lua_code_style_guidelines) code style guidelines.
     - Check your code works as expected and document any changes to the Lua API.
 
-3. Commit & [push](https://help.github.com/articles/pushing-to-a-remote/) your changes to a new branch (not `master`, one change per branch)
+4. Commit & [push](https://help.github.com/articles/pushing-to-a-remote/) your changes to a new branch (not `master`, one change per branch)
     - Commit messages should:
         - Use the present tense
-        - Have a title which begins with a capital letter
-        - Be descriptive. (e.g. no `Update init.lua` or `Fix a problem`)
-        - Have a first line with less than *80 characters* and have a second line that is *empty*
-
-4. Once you are happy with your changes, submit a pull request.
+        - Be descriptive (e.g. no `Update init.lua` or `Fix a problem`)
+    - The first line should:
+        - Start with a capital letter
+        - Be a compact summary of the commit
+        - Have less than 70 characters
+        - Have no full stop at the end
+        - Not contain a Github issue or pull request number
```

Comment:
i don't agree with that, it permits to have directly the issue or PR fixed, it's the most useful place.

Reply:
It seems better to have it in the later lines, a thread number is not descriptive and takes up precious room in the first line. Note, such an included number is not displayed as a link, and will be next to the link added by Github, so visually messy and a little confusing.

Reply:
But, not important, so ok.